### ISSUE TITLE
Add support for psql JSON operators

### ISSUE THREAD

**Issue by Fr1m3n** [mikro-orm/mikro-orm#4678]
Hello

Currently, only 2 JSON operators are [supported](https://mikro-orm.io/docs/query-conditions/#comparison) in mikro-orm - `@>` and `<@` (exclusively for psql). Postgresql has a large set of [operators](https://www.postgresql.org/docs/10/functions-json.html) (Table 9.44) that are not supported in mikro-orm. In particular, I was faced with the need to use the operator `?|`.

I assume a solution similar to the existing one with the operators `$contains` and `$contained`.

**Comment by B4nan** [comment]
PR welcome

**Comment by sslava** [comment]
Hey @B4nan, i'd like to implement support of `?`, `?|` and `?&` operators. 
What names do you think would suit best?
- `?` — `$haskey`
- `?&` — `$haskeys`
- `?|`  — `$hassomekeys` 

```ts
  await orm.em.findAll(User, {
    where: {
      books: { parameters: { $haskey: 'seasons' } },
    },
    populate: ["books"],
  });

  await orm.em.findAll(User, {
    where: { books: { parameters: { $haskeys: ['seasons', 'pages'] } } },
  });

  await orm.em.findAll(User, {
    where: { books: { parameters: { $hassomekeys: ['seasons', 'pages'] } } },
  });
  
```

Maybe you could propose better names?

**Comment by B4nan** [comment]
Let's at least use `pascalCase`, so `$hasKey` etc.

**Comment by sslava** [comment]
Cool, will fix the PR soon

**Comment by sslava** [comment]
@B4nan thanks for the reply. Fixed my [PR](https://github.com/mikro-orm/mikro-orm/pull/5366) with new naming

**Comment by B4nan** [comment]
I am closing this one since it's too broad and the mentioned operator is now included thanks to #5366.

If there are more operators missing, PR for those is still welcome, it should be hopefully pretty straightforward.
